src/main/java/net/snowflake/ingest/streaming/internal/ClientBufferParameters.java

```diff
@@ -89,6 +95,8 @@ public ClientBufferParameters(SnowflakeStreamingIngestClientInternal clientInter
         clientInternal != null
             ? clientInternal.getInternalParameterProvider().isEnableValuesCount()
             : InternalParameterProvider.ENABLE_VALUES_COUNT_DEFAULT;
+    this.enableDictionaryEncoding =
```

Comment:
Not very satisfied by how there's not a proliferation of places that control runtime behavior :( 
There's ParameterProvider, internalParameterProvider, clientBuferParameters, and some completely unnecesary defaulting for when clientInternal is null (it should never be not even in tests).

Tried to come up with a better answer but nothing that sticks and is relevant to your current PR.

As a followup whenever you have time i'd suggest first removing the null defaulting, remove the "isTestMode" thing from everywhere (fully doable after a round of changes I did for the ExternalVolume class). Lets see how things look after these two!

Reply:
Added [jira](https://snowflakecomputing.atlassian.net/browse/SNOW-1770243). Will merge this after the #874